pkg/workflows/wasm/host/module.go

```diff
@@ -580,9 +579,46 @@ func runWasm[I idMessage, O proto.Message](
 }
 
 func (m *module) SetCapabilityExecutor(handler CapabilityExecutor) error {
+	callCapAsync := toCallCapAsync(m.capabilityStore, handler)
+	if err := m.linker.FuncWrap(
+		"env",
+		"call_capability",
+		createCallCapFn(m.cfg.Logger, m.executeStore, callCapAsync),
+	); err != nil {
+		return fmt.Errorf("error wrapping callcap func: %w", err)
+	}
 	return errors.New("not implemented")
 }
 
+func toCallCapAsync(
+	capStore *store[*sdkpb.CapabilityResponse],
+	handler CapabilityExecutor,
+) func(ctx context.Context, req *sdkpb.CapabilityRequest) ([sdk.IdLen]byte, error) {
+	return func(ctx context.Context, req *sdkpb.CapabilityRequest) ([36]byte, error) {
+		callId := uuid.New()
+		idBuffer := make([]byte, sdk.IdLen)
+		copy(idBuffer, callId.String())
+
+		go func() {
+			resp, err := handler.CallCapability(ctx, req)
+			if err != nil {
+				resp = &sdkpb.CapabilityResponse{
+					Response: &sdkpb.CapabilityResponse_Error{
+						Error: err.Error(),
+					},
+				}
+			}
+			// TODO: Delete id on await
+			capStore.add(
```

Comment:
why do you need a capstore? You don't need to get context at runtime. Can/should you use a channel so you can read the response async later?